spec/factories/application_choice.rb

```diff
@@ -275,6 +275,14 @@
       end
     end
 
+    trait :with_changed_course do
+      after(:build) do |choice, _evaluator|
+        other_course = create(:course, provider: choice.course_option.course.provider)
+        choice.course_option_id = create(:course_option, course: other_course).id
```

Comment:
I think the current course option should also be set to the other course

Reply:
🆗